src/v/cluster/data_migration_backend.cc

```diff
@@ -1434,7 +1444,23 @@ backend::get_work_scope(const migration_metadata& metadata) {
     return std::visit(
       [&metadata](const auto& migration) {
           migration_direction_tag<std::decay_t<decltype(migration)>> tag;
-          return get_work_scope(tag, metadata);
+          auto scope = get_work_scope(tag, metadata);
+          if (migration.auto_advance && !scope.sought_state) {
+              switch (metadata.state) {
+              case state::planned:
+                  scope.sought_state = state::preparing;
+                  break;
+              case state::prepared:
+                  scope.sought_state = state::executing;
+                  break;
+              case state::executed:
+                  scope.sought_state = state::cut_over;
+                  break;
+              default:
+                  vassert(false, "Failed to automate state transition");
```

Comment:
maybe we can log here what was the state and overall context to help debugging such situations. 
nit: can we make the case list exhaustive and use `[[fallthrough]]` annotation